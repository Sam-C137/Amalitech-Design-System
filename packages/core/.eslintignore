dist
loader
www
**/*.e2e.ts
**/*.spec.tsx
**/*.spec.ts
