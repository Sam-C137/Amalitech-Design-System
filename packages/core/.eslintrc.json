{
  "parserOptions": {
    "project": "./tsconfig.json"
  },
  "extends": [
    "plugin:@stencil-community/recommended"
  ],
  "ignorePatterns": [
    "**/*.stories.tsx",
    "**/*.spec.*",
    "**/*.e2e.*"
  ],
  "rules": {
    "react/jsx-no-bind": "off"
  }
}
