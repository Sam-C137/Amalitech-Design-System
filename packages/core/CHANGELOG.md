# Change Log

All notable changes to this project will be documented in this file.
See [Conventional Commits](https://conventionalcommits.org) for commit guidelines.

## [0.8.3](https://github.com/Amali-Tech/Amalitech-Design-System/compare/v0.8.2...v0.8.3) (2024-07-18)


### Bug Fixes

* min max value attribute changes ([8276d9d](https://github.com/Amali-Tech/Amalitech-Design-System/commit/8276d9dd0d176071116497f01c12c92caecd5706))





## [0.8.2](https://github.com/Amali-Tech/Amalitech-Design-System/compare/v0.8.1...v0.8.2) (2024-06-02)

**Note:** Version bump only for package @amalitech-design-system/core





## [0.8.1](https://github.com/Amali-Tech/Amalitech-Design-System/compare/v0.8.0...v0.8.1) (2024-06-01)


### Bug Fixes

* tests ([05989c4](https://github.com/Amali-Tech/Amalitech-Design-System/commit/05989c4eaa674a902264d87c3044b6eb91794698))





# [0.8.0](https://github.com/Amali-Tech/Amalitech-Design-System/compare/v0.7.0...v0.8.0) (2024-05-27)


### Features

* add double border to radio ([21d478d](https://github.com/Amali-Tech/Amalitech-Design-System/commit/21d478d51df6c38aa88b8a1b429b76baa7371040))





# [0.7.0](https://github.com/Amali-Tech/Amalitech-Design-System/compare/v0.6.2...v0.7.0) (2024-05-14)


### Features

* add clear method to select ([689be3d](https://github.com/Amali-Tech/Amalitech-Design-System/commit/689be3df7d4c3f895ef8b2f11fc8bc24a2f78ab3))
* create select value accessor directive ([d5957c8](https://github.com/Amali-Tech/Amalitech-Design-System/commit/d5957c85d858f02ec8506671ae7ac142421a34c5))





## [0.6.2](https://github.com/Amali-Tech/Amalitech-Design-System/compare/v0.6.1...v0.6.2) (2024-05-06)

**Note:** Version bump only for package @amalitech-design-system/core





## [0.6.1](https://github.com/Amali-Tech/Amalitech-Design-System/compare/v0.6.0...v0.6.1) (2024-05-06)


### Bug Fixes

* removed default switch margin ([503f85d](https://github.com/Amali-Tech/Amalitech-Design-System/commit/503f85da2f380b6b7a89e3a46015ea367d67a2ce))





# [0.6.0](https://github.com/Amali-Tech/Amalitech-Design-System/compare/v0.5.0...v0.6.0) (2024-04-30)


### Bug Fixes

* pagination spacing ([00c99e4](https://github.com/Amali-Tech/Amalitech-Design-System/commit/00c99e4e17edf1f4d145135d8f1da444f6cb6ed6))


### Features

* exposed totalPagesCount in pagination ([4105737](https://github.com/Amali-Tech/Amalitech-Design-System/commit/4105737d72ba70a4bc0b9a7fa67dca5e73c2bc5b))





# [0.5.0](https://github.com/Amali-Tech/Amalitech-Design-System/compare/v0.4.0...v0.5.0) (2024-04-28)


### Features

* utilities align self ([ab106b6](https://github.com/Amali-Tech/Amalitech-Design-System/commit/ab106b6f3752e20155c9b307d89cbcc4b12c79ec))





# [0.4.0](https://github.com/Amali-Tech/Amalitech-Design-System/compare/v0.3.7...v0.4.0) (2024-04-28)


### Features

* add z-index to utilities ([75efd78](https://github.com/Amali-Tech/Amalitech-Design-System/commit/75efd78fbf0a22ce34139c7569a5ee894faaa088))





## [0.3.7](https://github.com/Amali-Tech/Amalitech-Design-System/compare/v0.3.6...v0.3.7) (2024-04-27)


### Bug Fixes

* date picker open method: ([710e97b](https://github.com/Amali-Tech/Amalitech-Design-System/commit/710e97b29a3538231ada47ffddd7314be1766559))





## [0.3.6](https://github.com/Amali-Tech/Amalitech-Design-System/compare/v0.3.5...v0.3.6) (2024-04-24)


### Bug Fixes

* centered the checkbox in it's label ([e7fbbaa](https://github.com/Amali-Tech/Amalitech-Design-System/commit/e7fbbaaf3f79a6d0185777a0ce52a44a6b6f84e8))






## [0.3.5](https://github.com/Amali-Tech/Amalitech-Design-System/compare/v0.3.4...v0.3.5) (2024-04-15)


### Bug Fixes

* build ([12140c9](https://github.com/Amali-Tech/Amalitech-Design-System/commit/12140c95e8be561d9f26d67c15b71c2beb50dee0))
* build ([a38dd2e](https://github.com/Amali-Tech/Amalitech-Design-System/commit/a38dd2e615bedc32ab7a68ce841a9b04e4f8267b))
* card tests ([5c99406](https://github.com/Amali-Tech/Amalitech-Design-System/commit/5c99406dc9796839ee06326fdaf2b8471e2d554d))
* starting with a value ([fb24ce3](https://github.com/Amali-Tech/Amalitech-Design-System/commit/fb24ce3cf7510dc5cca9c868aa03b48e3aa8acdb))
* tests ([373434e](https://github.com/Amali-Tech/Amalitech-Design-System/commit/373434e70de883671a1f828d1cbda137d717978a))
* tests ([eb11522](https://github.com/Amali-Tech/Amalitech-Design-System/commit/eb11522d23bf5b31c41228ba9f522998641c677f))
* tests ([e44f72d](https://github.com/Amali-Tech/Amalitech-Design-System/commit/e44f72d42a5b9fc587edfbeecafd3c1dc89b799a))
* tests ([799a98c](https://github.com/Amali-Tech/Amalitech-Design-System/commit/799a98cf03b6c008f749fdb754187abcca9cface))
* tests ([fcb7f47](https://github.com/Amali-Tech/Amalitech-Design-System/commit/fcb7f47051af324d7d5ea1c0a285f5f2645788be))
* tests ([66ea8f8](https://github.com/Amali-Tech/Amalitech-Design-System/commit/66ea8f888460610f550d88b51f1d4ec870444072))






## [0.3.4](https://github.com/Amali-Tech/Amalitech-Design-System/compare/v0.3.3...v0.3.4) (2024-04-05)

**Note:** Version bump only for package @amalitech-design-system/core






## [0.3.3](https://github.com/Amali-Tech/Amalitech-Design-System/compare/v0.3.2...v0.3.3) (2024-04-04)

**Note:** Version bump only for package @amalitech-design-system/core






## [0.3.2](https://github.com/Amali-Tech/Amalitech-Design-System/compare/v0.3.1...v0.3.2) (2024-04-03)

**Note:** Version bump only for package @amalitech-design-system/core





## [0.3.1](https://github.com/Amali-Tech/Amalitech-Design-System/compare/v0.3.0...v0.3.1) (2024-04-03)


### Bug Fixes

* tests ([a1ca9e2](https://github.com/Amali-Tech/Amalitech-Design-System/commit/a1ca9e26ce368e9d26b114605ad3869ccaa5dd8d))






# [0.3.0](https://github.com/Amali-Tech/Amalitech-Design-System/compare/v0.2.2...v0.3.0) (2024-04-02)


### Features

* form submission for input & button ([2cc64d6](https://github.com/Amali-Tech/Amalitech-Design-System/commit/2cc64d69378de5a3ec3e7c9da20f53e36e3a5dcc))






## [0.2.2](https://github.com/Amali-Tech/Amalitech-Design-System/compare/v0.2.1...v0.2.2) (2024-03-28)

**Note:** Version bump only for package @amalitech-design-system/core
