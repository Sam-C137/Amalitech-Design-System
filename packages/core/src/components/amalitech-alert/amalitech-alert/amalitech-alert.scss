@keyframes reveal {
  0% {
    opacity: 0;
    transform: translateY(1rem)
  }
}

@keyframes hide {
  100% {
    opacity: 0;
    transform: translateY(-1rem);
    visibility: hidden;
  }
}

:host {
  display: block;
  position: absolute;
  font-family: "Work Sans", sans-serif;
  z-index: var(--z-index-notification-manager, 100);
  pointer-events: none;
}

:host([name='global']) {
  position: fixed;
  width: 300px;
}

.amalitech-alert {
  pointer-events: none;
  display: flex;
  flex-direction: column;
  margin: var(--spacing-04, 0.75rem);

  .alert {
    animation: reveal 0.5s ease-in;
    pointer-events: auto;
  }

  .alert.hidden {
    display: none;
  }

}

:host([position="bottom-right"]) {
  bottom: 0;
  right: 0;

  .amalitech-alert {
    align-items: flex-end;
  }
}

:host([position="bottom-left"]) {
  bottom: 0;
  left: 0;

  .amalitech-alert {
    align-items: flex-start;
  }
}

:host([position="top-left"]) {
  top: 0;
  left: 0;

  .amalitech-alert {
    align-items: flex-start;
  }
}

:host([position="top-right"]) {
  top: 0;
  right: 0;

  .amalitech-alert {
    align-items: flex-end;
  }
}

