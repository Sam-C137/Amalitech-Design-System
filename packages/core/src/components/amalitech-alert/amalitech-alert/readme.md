# amalitech-alert



<!-- Auto Generated Below -->


## Properties

| Property   | Attribute  | Description           | Type                                                           | Default          |
| ---------- | ---------- | --------------------- | -------------------------------------------------------------- | ---------------- |
| `name`     | `name`     |                       | `string`                                                       | `'global'`       |
| `position` | `position` | position of the alert | `"bottom-left" \| "bottom-right" \| "top-left" \| "top-right"` | `'bottom-right'` |


## Dependencies

### Depends on

- [amalitech-alert-content](../amalitech-alert-content)

### Graph
```mermaid
graph TD;
  amalitech-alert --> amalitech-alert-content
  style amalitech-alert fill:#f9f,stroke:#333,stroke-width:4px
```

----------------------------------------------

*Built with [StencilJS](https://stenciljs.com/)*
