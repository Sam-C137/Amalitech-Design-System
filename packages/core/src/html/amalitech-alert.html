<!DOCTYPE html>
<html lang="en">
  <head>
    <meta charset="utf-8" />
    <meta name="viewport" content="width=device-width, initial-scale=1.0" />
    <title>amalitech-alert Preview</title>
    <link rel="preconnect" href="https://fonts.googleapis.com" />
    <link rel="preconnect" href="https://fonts.gstatic.com" crossorigin="" />
    <link
      href="https://fonts.googleapis.com/css2?family=Work+Sans:ital,wght@0,100;0,200;0,300;0,400;0,500;0,600;0,700;0,800;0,900;1,100;1,200;1,300;1,400;1,500;1,600;1,700;1,800;1,900&amp;display=swap"
      rel="stylesheet"
    />
    <script type="module" src="/build/amalitech-design-system.esm.js"></script>
    <script nomodule src="/build/build/amalitech-design-system.js"></script>
    <link rel="stylesheet" href="/build/amalitech-design-system.css" />
    <script src="https://code.iconify.design/2/2.1.0/iconify.min.js"></script>
    <link rel="stylesheet" href="https://cdnjs.cloudflare.com/ajax/libs/font-awesome/4.7.0/css/font-awesome.min.css" />
    <style>
      amalitech-alert {
        margin-bottom: 20px;

        & > p {
          font-weight: bold;
        }
      }
    </style>
  </head>
  <body>
    <amalitech-alert name="demo-notification-manager" position="top-right" class="hydrated">
      <amalitech-alert-content state="success" class="hydrated">
        <div slot="title">...</div>
        <div slot="subtitle">...</div>
      </amalitech-alert-content>
    </amalitech-alert>
    <amalitech-button variant="secondary" size="medium" type="button" id="notification-btn" icon-position="right" class="hydrated"> Show Alert</amalitech-button>

    <amalitech-alert variant="success" closeable close-secs="100" auto-close>
      <p>Some message</p>
    </amalitech-alert>

    <amalitech-alert variant="warning" closeable close-secs="100" auto-close>
      <p>Some message</p>
    </amalitech-alert>

    <amalitech-alert variant="error" closeable close-secs="100" auto-close>
      <p>Some message</p>
    </amalitech-alert>

    <amalitech-alert variant="info" closeable close-secs="100" auto-close>
      <p>Some message</p>
    </amalitech-alert>

    <script>
      document.getElementById('notification-btn').addEventListener('click', () => {
        window.dispatchEvent(
          new CustomEvent('alert:alert-notification', {
            detail: {
              target: 'demo-notification-manager',
              title: 'Welcome',
              state: 'info',
              subtitle: 'This is boilerplate code for Vanilla JS + CDN.',
            },
          }),
        );
      });
    </script>
  </body>
</html>
