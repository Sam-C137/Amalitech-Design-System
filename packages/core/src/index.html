<!DOCTYPE html>
<html dir="ltr" lang="en">
  <head>
    <meta charset="utf-8" />
    <meta name="viewport" content="width=device-width, initial-scale=1.0" />
    <title>Stencil Component Starter</title>
    <script type="module" src="/build/amalitech-design-system.esm.js"></script>
    <script nomodule src="/build/amalitech-design-system.js"></script>
    <link rel="stylesheet" href="/build/amalitech-design-system.css" />
    <style>
      h1 {
        text-align: center;
        font-style: normal;
        line-height: 2.5rem;
        font-family: var(--amalitech-font-variant-body-family);
        color: var(--amalitech-color-neutral-700);
        font-size: var(--amalitech-font-variant-heading-3-size);
        font-weight: var(--amalitech-font-weight-bold);
        margin: 0;
        padding: 0;
      }
    </style>
  </head>

  <body>
    <h1>This is ADS</h1>
  </body>
</html>
