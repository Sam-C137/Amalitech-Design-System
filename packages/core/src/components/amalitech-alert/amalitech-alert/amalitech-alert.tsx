import { Component, ComponentInterface, Element, h, Host, Listen, Prop, State } from '@stencil/core';
import { AlertTypes } from '../../../global/types';

const getNotificationIndex = (() => {
  let counter = 1;
  return function () {
    return `${counter++}`;
  };
})();

@Component({
  tag: 'amalitech-alert',
  styleUrl: 'amalitech-alert.scss',
  shadow: true,
})
export class AmalitechAlert implements ComponentInterface {
  @Element() elm!: HTMLAmalitechAlertElement;

  /**
   * 
   */
  @Prop({ reflect: true }) name: string = 'global';

  /**
   * position of the alert
   */
  @Prop() position: 'top-right' | 'top-left' | 'bottom-right' | 'bottom-left' = 'bottom-right';

  
  @State() alerts: any = [];

  @Listen('alert:alert-notification', { target: 'window' })
  listenNotification(evt: any) {
    if ((evt.detail.target === this.name || this.name === 'global') && !evt.detail.procced) {
      evt.preventDefault();
      evt.stopPropagation();
      evt.detail.procced = true;
      const alert: AlertTypes = {
        id: getNotificationIndex(),
        type: 'notification',
        title: evt.detail.title,
        subtitle: evt.detail.subtitle,
        message: evt.detail.message,
        messageType: evt.detail.messageType,
        state: evt.detail.state,
        hide: false,
      };
      this.alerts = this.alerts.concat([alert]).filter((n: any) => !n.hide);
      setTimeout(() => {
        alert.hide = true;
        this.alerts = [...this.alerts];
      }, 5000);
    }
  }

  private renderAlerts(alert: AlertTypes) {
    if (alert.type !== null) {
      switch (alert.type) {
        case 'notification':
          return (
            <amalitech-alert-content id={`${alert.id}`} state={alert.state}>
              <div innerHTML={alert.title} slot="title" />
              <div innerHTML={alert.subtitle} slot="subtitle" />
            </amalitech-alert-content>
          );
        default:
          return null;
      }
    }
  }

  render() {
    return (
      <Host>
        <div class={{ 'amalitech-alert': true, [`position-${this.position}`]: true }}>
          {this.alerts.map((alert: AlertTypes) => (
            <div class={{ alert: true, hidden: alert.hide }}>{this.renderAlerts(alert)}</div>
          ))}
        </div>
      </Host>
    );
  }
}
