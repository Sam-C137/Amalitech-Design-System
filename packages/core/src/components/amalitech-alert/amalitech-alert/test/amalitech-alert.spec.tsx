import { newSpecPage } from '@stencil/core/testing';
import { AmalitechAlert } from '../amalitech-alert';

describe('amalitech-alert', () => {
  it('renders alerts correctly', async () => {
    const page = await newSpecPage({
      components: [AmalitechAlert],
      html: '<amalitech-alert></amalitech-alert>',
    });
    expect(page.root).toMatchSnapshot();
  });

  it('adds alerts correctly', async () => {
    const page = await newSpecPage({
      components: [AmalitechAlert],
      html: '<amalitech-alert></amalitech-alert>',
    });

    const alertData = {
      title: 'Test Title',
      subtitle: 'Test Subtitle',
      message: 'Test Message',
      messageType: 'info',
      state: 'open',
    };

    window.dispatchEvent(new CustomEvent('alert:alert-notification', { detail: alertData }));
    await page.waitForChanges();

    expect(page.rootInstance.alerts).toHaveLength(1);
    expect(page.root).toMatchSnapshot();
  });
});
