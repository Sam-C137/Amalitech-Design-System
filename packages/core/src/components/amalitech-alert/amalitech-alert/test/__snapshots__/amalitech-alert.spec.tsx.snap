// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`amalitech-alert adds alerts correctly 1`] = `
<amalitech-alert name="global">
  <mock:shadow-root>
    <div class="amalitech-alert position-bottom-right">
      <div class="alert">
        <amalitech-alert-content id="1" state="open">
          <div slot="title">
            Test Title
          </div>
          <div slot="subtitle">
            Test Subtitle
          </div>
        </amalitech-alert-content>
      </div>
    </div>
  </mock:shadow-root>
</amalitech-alert>
`;

exports[`amalitech-alert renders alerts correctly 1`] = `
<amalitech-alert name="global">
  <mock:shadow-root>
    <div class="amalitech-alert position-bottom-right"></div>
  </mock:shadow-root>
</amalitech-alert>
`;
